src/monitoring_service/service.py

```diff
@@ -197,8 +191,22 @@ def _trigger_scheduled_events(self) -> None:
         for scheduled_event in triggered_events:
             event = scheduled_event.event
 
-            handle_event(event, self.context)
-            self.context.database.remove_scheduled_event(scheduled_event)
+            try:
+                handle_event(event, self.context)
+            except TransactionTooEarlyException:
+                log.debug(
+                    "Event executed too early. "
+                    "Retry later and don't try any other scheduled events right now.",
+                    handled_event=event,
+                )
+                # When the scheduled event with the lowest timestamp fails with
```

Comment:
From the code in the database module I'm not sure: is the event list actually sorted by timestamps?
In any case I would probably help to also express this somewhere. At least in the docs of the database function or even in the function name itself.

Reply:
Good catch! We really need to sort explicitly now. I'll push a commit for that.

Reply:
Good catch! We really need to sort explicitly now. I'll push a commit for that.

Reply:
Good catch! We really need to sort explicitly now. I'll push a commit for that.

Reply:
Got resolved with [e66c720](https://github.com/raiden-network/raiden-services/pull/1158/commits/e66c7204fb65678bcff63a51a35a789f1ca10bea) 👍🏾